### PR TITLE
WFLY-8207, WFLY-6973 JGroups 4.0, Infinispan 9.1 upgrade

### PR DESCRIPTION
**Pull request by pferraro**
Includes prerequisite/dependent jiras:
WFLY-9229 Allow multiple CommandDispatcher instances per service (breaks distributed shared sessions, addressed by fixes for WFLY-9344 & WFLY-9428)
WFLY-9271 Create CommandDispatcher-based BroadcastEndpoint
WFLY-9344 Outsource distributed HttpSession/SFSB passivation to Infinispan
WFLY-9428 Fix premature expiration of shared distributed web sessions

https://issues.jboss.org/browse/WFLY-8207
https://issues.jboss.org/browse/WFLY-6973
https://issues.jboss.org/browse/WFLY-9229
https://issues.jboss.org/browse/WFLY-9271
https://issues.jboss.org/browse/WFLY-9428
https://issues.jboss.org/browse/WFLY-9344

This PR includes a commit from: https://github.com/wildfly/wildfly/pull/10654

@jmesnil Please review WFLY-9271 commit

### CODE REVIEW

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/broadcast/QueueBroadcastManager.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * @author Paul Ferraro
+ */
+public class QueueBroadcastManager implements BroadcastManager {
+    private final BlockingQueue<byte[]> broadcasts = new LinkedBlockingDeque<>();
```
[the remark this answers is not in the document]
Comment (pferraro) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/broadcast/QueueBroadcastManager.java]: This is not a new requirement.  QueueBroadcastManager performs the same role as the existing JGroupsReceiver.  The implementations are identical:
https://github.com/apache/activemq-artemis/blob/master/artemis-core-client/src/main/java/org/apache/activemq/artemis/api/core/jgroups/JGroupsReceiver.java

[the remark this answers is not in the document]
Comment (jmesnil) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/broadcast/QueueBroadcastManager.java]: ah ok, you also replaced some bits from Artemis. looks fine then

**Review by jmesnil — COMMENTED** [review]
I reviewed the changes in the `messaging-activemq` subsystem and they look good.

**Comment by pferraro** [comment]
I'm going to split this up into smaller PRs in an attempt to hasten the review/merge process.

**Comment by pferraro** [comment]
@jmesnil FYI - I've split out the messaging-activemq changes into a separate PR: https://github.com/wildfly/wildfly/pull/10678

**Pull request closed without merging**